### PR TITLE
backup-manager: try to directly get the backup size from backupmeta before calculate the backup size

### LINKED ISSUE #5407: operator parse br backupmeta v2 get incorrect files storage size
## Bug Report

**What version of Kubernetes are you using?**

**What version of TiDB Operator are you using?**

```
TiDB Operator Version: version.Info{GitVersion:"v1.6.0-alpha.8", GitCommit:"707e8805b83d1dc0f036f3eceac70349a6111ab9", GitTreeState:"clean", BuildDate:"2023-09-02T00:08:15Z", GoVersion:"go1.19.12", Compiler:"gc", Platform:"linux/amd64"}
```

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

**What's the status of the TiDB cluster pods?**

**What did you do?**

snapshot backup cluster

**What did you expect to see?**
operator parse `br` backupmeta v2 get correct files storage size

**What did you see instead?**
https://github.com/pingcap/tidb-operator/blob/master/cmd/backup-manager/app/util/util.go#L354
get incorrect storage size

img_v3_025d_4b2bfbd2-f1ec-4d66-a781-6c3e9ab0168g

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

The current backup-manager cannot parse the backupmetaV2.

Close #5407 
### What is changed and how does it work?

try to directly get the backup size from backupmeta before calculate the backup size. Notice that if the br supports backupemtaV2, it also supports to record the backup size into the backupmeta.
### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Comment by Leavrth** [comment]
blocken by https://github.com/pingcap/tidb/pull/48772

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ module github.com/pingcap/tidb-operator
 go 1.21
 
 require (
-	cloud.google.com/go/storage v1.10.0
+	cloud.google.com/go/storage v1.30.1
```
Comment (BornChanger) [inline go.mod:10]: Why there is go.mod change?

Comment (Leavrth) [reply go.mod:10]: upgrade to `kvproto v0.0.0-20231122054644-fb0f5c2a0a10`,
which imports `cloud.google.com/go v0.110.10`,
which imports `cloud.google.com/go/storage v1.30.1`

`go.mod` use the newest version of the package.

**Review by BornChanger — APPROVED** [review]

**Review by WangLe1321 — APPROVED** [review]

**Pull request merged**